### ISSUE TITLE
control duration \delta t in robots.cpp?

### ISSUE THREAD

**Issue by ucas-zihaowang** [IMRCLab/db-CBS#124]
Hello, @akmaralAW  In the db-CBS algorithm, the control duration size \delta t is set to 0.1s, as can be seen in the code robots.cpp. I would like to know if the current db-CBS algorithm supports changing \delta t to 0.5s in robots.cpp. Would I need to regenerate the motion primitives .msgpack file? Thanks!

**Comment by akmaralAW** [comment]
Hi @ucas-zihaowang,
Yes, we used dt=0.1, in case you want to change it to dt-0.5, then you have to generate new motion primitives. Generating new motion primitives is fast, you can use [dynoplan](https://github.com/quimortiz/dynoplan), it has small turorial on how to do that. I hope it helps you.

**Comment by ucas-zihaowang** [comment]
@akmaralAW. Excuse me, I'm currently confused about the db-A* algorithm paper and would like to clarify some related questions with you.
(1) In the db-A* paper, regarding the motion primitive generation in section IV-C, the third step involves splitting the motion into multiple segments according to the desired length. Does this desired length refer to the trajectory length or the time duration size? According to in db-CBS project gen_motion_primitive_komo.py lines 76-84, it is split based on the Euclidean distance being greater than 0.5. This suggests that the motion primitive generation is unrelated to the time duration size.

(2) In the implementation of db-A*, after debugging, the time duration of the motion primitives during the db-A* search is not 0.1s, because the cost of expanding each motion primitive is not 0.1. Instead, many intermediate states are inserted after the entire search is completed, and these intermediate states have a time duration of 0.1s, as can be verified in line 630 of db_astar.hpp.

In summary, the motion primitives used in db-A* search are not those with a time duration of 0.1s, but rather those divided by Euclidean length. Therefore, I think that setting the time duration to 0.5s should not require regenerating the motion primitives.

Maybe my thoughts are wrong, thanks for your help.

**Comment by akmaralAW** [comment]
hi @ucas-zihaowang,

1) Regarding the link between dt=0.1 (step size) and motion primitives: 
each motion primitive is a short trajectory = sequence of states.  We represent the motion primitive with states, actions. 
For each dynamics we use Euler integration to get the state of the robot over time (Eq. 2 in db-A* paper, or check src/robots.cpp file, propagate() function for each robot dynamics). The length of each motion primitive is equal to number_of_actions * dt.

2) The first answer is relevant to this one as well. 

I hope it helps understand the paper/implementation better.

**Comment by ucas-zihaowang** [comment]
@akmaralAW Thanks for your reply!!!  I have fully understood what you explained.
